### DISCUSSION TITLE
XOR을 이용한 swap 기법

### DISCUSSION THREAD

**Discussion by wogha95** [DaleStudy/leetcode-study#423] [자료공유]
신기하면서 swap 기법을 다양하게 생각해볼 수 있는 계기가 되었습니다.
핵심 부분만 작성해두었고 자세한 내용은 하단 링크를 참고 부탁드립니다. ;)

```javascript
// [Before] x = 2, y = 5

x = x ^ y;
y = x ^ y;
x = x ^ y;

// [After] x = 5, y = 2
```

- https://driip.me/f59bdfd5-eeaf-4fad-9bb1-27923f38b16a

**Comment by DaleSeo** [comment]
어떤 과정을 거쳐서 swap이 되는지 공유해주신 글을 보고 가까스로 이해했네요. 그런데 도대체 이 swap 기법을 언제 어디서 활용하면 좋을지는 잘 모르겠습니다. 코딩 테스트에서도 그렇고 실무에서도 요렇게 코딩하면 괴짜 취급 받지 않을까요? 😂
